packages/cardano-services/src/Asset/TypeOrmNftMetadataService.ts

```diff
@@ -0,0 +1,42 @@
+import { Asset, Cardano } from '@cardano-sdk/core';
+import { AssetPolicyIdAndName, NftMetadataService } from './types';
+import { Logger } from 'ts-log';
+import { NftMetadataEntity } from '@cardano-sdk/projection-typeorm';
+import { TypeormProvider, TypeormProviderDependencies } from '../util/TypeormProvider';
+
+export class TypeOrmNftMetadataService extends TypeormProvider implements NftMetadataService {
+  #logger: Logger;
+
+  // Accept database connection string
+  constructor({ connectionConfig$, logger, entities }: TypeormProviderDependencies) {
+    super('TypeOrmNftMetadataService', { connectionConfig$, entities, logger });
+    this.#logger = logger;
+  }
+
+  async getNftMetadata(
+    assetInfo: AssetPolicyIdAndName & { assetId: Cardano.AssetId }
+  ): Promise<Asset.NftMetadata | null> {
+    return this.withDataSource(async (dataSource) => {
+      const queryRunner = dataSource.createQueryRunner();
+      const nftMetadataRepository = queryRunner.manager.getRepository(NftMetadataEntity);
+
+      this.#logger.debug('Querying database');
+      const asset = await nftMetadataRepository.findOneBy({
+        name: assetInfo.name,
+        parentAsset: { id: assetInfo.assetId }
```

Comment:
As the doc comment suggests, `parentAsset` is the reference token for cip68 assets. We're always interested in metadata for the user token (token that lives in user's wallet) in this context.
```suggestion
        userTokenAsset: { id: assetInfo.assetId }
```

Reply:
Fixed on 2c261b3